miner/scroll_worker.go

```diff
@@ -0,0 +1,945 @@
+		case res.OverflowingTx.IsL1MessageTx() &&
+			errors.Is(res.CCCErr, circuitcapacitychecker.ErrUnknown):
+			l1TxCccUnknownErrCounter.Inc(1)
+		case !res.OverflowingTx.IsL1MessageTx() &&
+			errors.Is(res.CCCErr, circuitcapacitychecker.ErrUnknown):
+			l2TxCccUnknownErrCounter.Inc(1)
+		}
+	}
+
+	if !w.isRunning() {
+		if res != nil && res.FinalBlock != nil {
+			w.updateSnapshot(res.FinalBlock)
+		}
+		w.currentPipeline = nil
+		return nil
+	}
+
+	if res == nil || res.FinalBlock == nil {
+		w.startNewPipeline(time.Now().Unix())
+		return nil
+	}
+	return w.commit(res)
+}
+
+// commit runs any post-transaction state modifications, assembles the final block
+// and commits new work if consensus engine is running.
+func (w *worker) commit(res *pipeline.Result) error {
+	defer func(t0 time.Time) {
+		l2CommitTimer.Update(time.Since(t0))
+	}(time.Now())
+
+	if res.CCCErr != nil {
+		commitReasonCCCCounter.Inc(1)
+	} else {
+		commitReasonDeadlineCounter.Inc(1)
+	}
+	commitGasCounter.Inc(int64(res.FinalBlock.Header.GasUsed))
+
+	block, err := w.engine.FinalizeAndAssemble(w.chain, res.FinalBlock.Header, res.FinalBlock.State,
+		res.FinalBlock.Txs, nil, res.FinalBlock.Receipts)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case w.taskCh <- &task{receipts: res.FinalBlock.Receipts, state: res.FinalBlock.State, block: block, createdAt: time.Now(),
+		accRows: res.Rows, nextL1MsgIndex: res.FinalBlock.NextL1MsgIndex}:
+		w.unconfirmed.Shift(block.NumberU64() - 1)
+		log.Info("Commit new mining work", "number", block.Number(), "sealhash", w.engine.SealHash(block.Header()),
+			"txs", res.FinalBlock.Txs.Len(),
+			"gas", block.GasUsed(), "fees", totalFees(block, res.FinalBlock.Receipts),
+			"elapsed", common.PrettyDuration(time.Since(w.currentPipelineStart)))
+	case <-w.exitCh:
+		log.Info("Worker has exited")
+	}
+
+	w.currentPipeline = nil
+	return nil
+}
+
+// copyReceipts makes a deep copy of the given receipts.
+func copyReceipts(receipts []*types.Receipt) []*types.Receipt {
+	result := make([]*types.Receipt, len(receipts))
+	for i, l := range receipts {
+		cpy := *l
+		result[i] = &cpy
+	}
+	return result
+}
+
+// postSideBlock fires a side chain event, only use it for testing.
+func (w *worker) postSideBlock(event core.ChainSideEvent) {
+	select {
+	case w.chainSideCh <- event:
+	case <-w.exitCh:
+	}
+}
+
+func (w *worker) onTxFailingInPipeline(txIndex int, tx *types.Transaction, err error) bool {
+	writeTrace := func() {
+		var trace *types.BlockTrace
+		var errWithTrace *pipeline.ErrorWithTrace
+		if w.config.StoreSkippedTxTraces && errors.As(err, &errWithTrace) {
+			trace = errWithTrace.Trace
+		}
+		rawdb.WriteSkippedTransaction(w.eth.ChainDb(), tx, trace, err.Error(),
+			w.currentPipeline.Header.Number.Uint64(), nil)
+	}
+
+	switch {
+	case errors.Is(err, core.ErrGasLimitReached) && tx.IsL1MessageTx():
+		// If this block already contains some L1 messages,
+		// terminate here and try again in the next block.
```

Comment:
But we don't really terminate here, for that we'd need to `return true`

Reply:
copy pasta error. fixed https://github.com/scroll-tech/go-ethereum/pull/735/commits/05bbd32e904c8447460d601619779e1c935987a4